packages/core/src/browser/storage-service.ts

```diff
@@ -76,4 +85,33 @@ export class LocalStorageService implements StorageService {
         const pathname = typeof window === 'undefined' ? '' : window.location.pathname;
         return `theia:${pathname}:${key}`;
     }
+
+    private ShowMessageNotification() {
+        const READ_INSTRUCTIONS_ACTION = "Read Instructions";
+        const ERROR_MESSAGE = `Your prefered browser local storage is almost full.
+            To be able to save your current workspace layout or data, you may need to free some space.
+            You can refer to Theia page for instructions.
+            on how to manually clean your browser local storage.`;
+        this.messageService.warn(ERROR_MESSAGE, READ_INSTRUCTIONS_ACTION).then(selected => {
+            if (READ_INSTRUCTIONS_ACTION === selected) {
+                window.open("https://github.com/theia-ide/theia/wiki/Cleaning-Local-Storage");
+            }
+        });
+    }
+
+    private testLocalStorage(): void {
+        const array = new Array(60000); // size: <array size> * 5 =  ~ 300K
+        const keyTest = this.prefix(`Test`);
+
+        if (this.storage !== this.fallbackStorage) {
+            try {
+                this.storage[keyTest] = JSON.stringify(array);
+                this.storage.removeItem(keyTest);
+            } catch (error) {
+                this.ShowMessageNotification();
+                this.storage.removeItem(keyTest);
```

Comment:
perhaps use a ``try catch finally`` since in both cases we will need to make sure that the test data is removed and we want to minimize duplicate code?

Reply:
Done, Use finally